kaitaistruct.py

```diff
@@ -393,10 +475,10 @@ def process_rotate_left(data, amount, group_size):
     # ========================================================================
 
     @staticmethod
-    def int_from_byte(v):
-        if PY2:
+    def int_from_byte(v, signed=False):
+        if PY_OLD:
             return ord(v)
-        return v
+        return int.from_bytes(v, signed=signed)
```

Comment:
`int_from_byte` is only a Python 2/3 compatibility method, for converting a 1-byte string into an integer byte value (needed because of API differences between Python 2 `str` and Python 3 `bytes`). I don't think it makes sense to add support for reading signed integers here.

In this state the implementation also doesn't work correctly on Python 2, because the `signed` paramter is ignored in the `PY2`/`PY_OLD` branch.

Reply:
Why not? We can use it as a service method to convert integers this way. Also 3.2 version check was added exactly for `from_bytes` method as it is not implemented prior.

Reply:
This method is meant to be used in contexts like this:

```python
byte_string = b"hello" # str in Python 2, bytes in Python 3
print(KaitaiStream.int_from_byte(byte_string[2]))
for byte in byte_string:
    print(KaitaiStream.int_from_byte(byte))
```

In Python 2, byte strings (`str`) are sequences of 1-byte strings. In Python 3, byte strings (`bytes`) are sequences of integers. `int_from_bytes` is used to get a consistent behavior on both versions, by converting Python 2's 1-byte strings into integers, and returning the argument unchanged on Python 3 (because it's already an integer).

If you need a way to read signed 1-byte integers, you can use the already existing `packer_s1` struct object.

Reply:
Sorry, it was really dumb of me. :facepalm: